tests/org.jboss.tools.examples.ui.bot.test/src/org/jboss/tools/examples/ui/bot/test/integration/EAPXPImportQuickstartsTest.java

```diff
@@ -0,0 +1,156 @@
+	public static final String SERVER_NAME = "Enterprise Application Platform";
+	public static final String BLACKLIST_FILE = "resources/servers/eap-blacklist";
+	public static final String BLACKLIST_ERRORS_REGEXES_FILE = "resources/servers/eap-xp-blacklist-test-errors-regexes.json";
+	public static final String URLS_FILE = "resources/servers/eap-xp-quickstarts-urls.json";
+	protected JSONObject urlsFileContents;
+
+	@Parameters(name = "{0}")
+	public static Collection<Quickstart> data() {
+		return createQuickstartsList();
+	}
+
+	@Parameter
+	public Quickstart qstart;
+
+	/*
+	 * Main tests. Imports quickstart as maven project, performs checks and deploy
+	 * on server.
+	 */
+	@Test
+	public void quickstartTest() {
+		runQuickstarts(qstart, SERVER_NAME, BLACKLIST_FILE, BLACKLIST_ERRORS_REGEXES_FILE);
+	}
+
+	/**
+	 * Checks whether is project deployed properly.
+	 * 
+	 * @param projectName
+	 * @param serverNameLabel
+	 */
+	@Override
+	public void checkDeployedProject(Quickstart qstart, String serverNameLabel) {
+		this.urlsFileContents = loadURLsForQuickstartsFromFile(URLS_FILE);
+		// Close Opened browser
+		closeBrowser();
+		ConsoleView console = new ConsoleView();
+		console.open();
+		new WaitUntil(new ConsoleHasText(console, "Deployed"), TimePeriod.LONG);
+
+		WebBrowserView browser = new WebBrowserView();
+		browser.open();
+		if (!(urlsFileContents == null
+				|| (!urlsFileContents.containsKey(qstart.getName()) && !urlsFileContents.containsKey("*")))) {
+
+			String quickstartURL = "";
+			if (urlsFileContents.containsKey(qstart.getName())) {
+				quickstartURL = (String) urlsFileContents.get(qstart.getName()).toString();
+			}
+			browser.openPageURL(quickstartURL);
+		}
+
+		try {
+			new WaitUntil(new BrowserIsnotEmpty(browser), TimePeriod.LONG);
+		} catch (WaitTimeoutExpiredException e) {
+			// try to refresh browser and wait one more time.
+			browser.refreshPage();
+			new WaitUntil(new BrowserIsnotEmpty(browser), TimePeriod.LONG);
+		}
+
+		assertNotEquals("", browser.getText());
+		browser.close();
+
+		checkConsoleForException();
+
+	}
+
+	public static JSONObject loadURLsForQuickstartsFromFile(String urlsFile) {
+		JSONObject urlsFileContents = null;
+
+		if (!urlsFile.isEmpty()) {
+			String pathToFile = "";
+			try {
+				pathToFile = new File(urlsFile).getCanonicalPath();
+				JSONParser parser = new JSONParser();
+				urlsFileContents = (JSONObject) parser.parse(new FileReader(pathToFile));
+			} catch (IOException ex) {
+				fail("urls file not found! Path is: " + pathToFile);
+			} catch (ParseException e) {
+				fail("ParseException: unable to parse file at path is: " + pathToFile);
+			}
+		}
+
+		return urlsFileContents;
+	}
+
+	class BrowserIsnotEmpty extends AbstractWaitCondition {
+
+		WebBrowserView browser;
+
+		public BrowserIsnotEmpty(WebBrowserView browser) {
+			this.browser = browser;
+		}
+
+		public boolean test() {
+			return !browser.getText().equals("");
```

Comment:
@jkopriva when does it happen that browser is empty? If I open new web browser window I can get html body basic tags. What is the situation when one can get empty text?

Reply:
This could happen, when quickstart is not correctly deployed or is deploying and the browser is loading.

Reply:
alright